nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/BulkRemoveEnrichmentAction.java

```diff
@@ -0,0 +1,151 @@
+import static org.nuxeo.lib.stream.computation.AbstractComputation.INPUT_1;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.OUTPUT_1;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ai.metadata.LabelSuggestion;
+import org.nuxeo.ai.metadata.SuggestionMetadataWrapper;
+import org.nuxeo.ai.services.DocMetadataService;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentNotFoundException;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.PropertyException;
+import org.nuxeo.ecm.core.bulk.action.computation.AbstractBulkComputation;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+/**
+ *  A bulk action to remove enrichment metadata
+ */
+public class BulkRemoveEnrichmentAction implements StreamProcessorTopology {
+
+    public static final String ACTION_NAME = "bulkEnrichRemove";
+
+    public static final String PARAM_MODEL = "modelId";
+
+    public static final String PARAM_XPATHS = "xpaths";
+
+    @Override
+    public Topology getTopology(Map<String, String> options) {
+
+        return Topology.builder()
+                       .addComputation(RemovalComputation::new,
+                                       Arrays.asList(INPUT_1 + ":" + ACTION_NAME, //
+                                                     OUTPUT_1 + ":" + STATUS_STREAM))
+                       .build();
+    }
+
+    public static class RemovalComputation extends AbstractBulkComputation {
+
+        private static final Logger log = LogManager.getLogger(RemovalComputation.class);
+
+        protected String modelId;
+
+        protected Set<String> xPaths;
+
+        public RemovalComputation() {
+            super(ACTION_NAME);
+        }
+
+        @Override
+        public void startBucket(String bucketKey) {
+            BulkCommand command = getCurrentCommand();
+            Serializable modelParam = command.getParam(PARAM_MODEL);
+            Serializable xParam = command.getParam(PARAM_XPATHS);
+            modelId = modelParam != null ? String.valueOf(modelParam) : null;
+            xPaths = xParam != null ? getSet(xParam) : Collections.emptySet();
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            DocMetadataService metadataService = Framework.getService(DocMetadataService.class);
+            for (String id : ids) {
+                try {
+                    DocumentModel doc = session.getDocument(new IdRef(id));
```

Comment:
instead of capturing exception you can check it via `CoreSession#exists`

Reply:
Would that then be less efficient because its making 2 calls?

Reply:
it's an interesting question. 
- Don't think so, first retrieving the doc should be O(1). second I'd assume that just the creation of exception is a relatively expensive process and it depends on how deep the stack trace is.

Reply:
Found the `loadDocuments` method instead